src/wp-includes/link-template.php

```diff
@@ -192,7 +192,7 @@ function get_permalink( $post = 0, $leavename = false ) {
 
 				$category_object = get_term( $category_object, 'category' );
 				$category        = $category_object->slug;
-				if ( $parent = $category_object->parent ) {
+				if ( $category_object->parent ) {
 					$category = get_category_parents( $parent, false, '/', true ) . $category;
```

Comment:
This change does not declare `$parent` used in line 196

Reply:
Agreed, `$parent` needs switching to `$category_object->parent`, I'll get that done later this week.